src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,76 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commanders = (0 until settings.numberOfShards)
```

Comment:
Here we can use getCommander (id) directly because the position of the record in the shard is determined by the part id.

Reply:
```suggestion
    val commander: EntityRef[Command] = getCommander(id)
```